modules/objectdb_profiler/objectdb_profiler_panel.cpp

```diff
@@ -0,0 +1,445 @@
+		view->show_snapshot(snapshot, diff);
+	}
+}
+
+void ObjectDBProfilerPanel::clear_snapshot() {
+	for (SnapshotView *view : views) {
+		view->clear_snapshot();
+	}
+	current_snapshot = nullptr;
+}
+
+void ObjectDBProfilerPanel::set_enabled(bool p_enabled) {
+	take_snapshot->set_text("Take ObjectDB Snapshot");
+	take_snapshot->set_disabled(!p_enabled);
+}
+
+void ObjectDBProfilerPanel::_snapshot_rmb(const Vector2 &p_pos, MouseButton p_button) {
+	if (p_button != MouseButton::RIGHT) {
+		return;
+	}
+	rmb_menu->clear(false);
+
+	rmb_menu->add_icon_item(get_editor_theme_icon(SNAME("Rename")), "Rename Snapshot", RC_MENU_OPERATIONS::RENAME);
+	rmb_menu->add_icon_item(get_editor_theme_icon(SNAME("Folder")), "Show In Folder", RC_MENU_OPERATIONS::SHOW_IN_FOLDER);
+	rmb_menu->add_icon_item(get_editor_theme_icon(SNAME("Remove")), "Delete Snapshot", RC_MENU_OPERATIONS::DELETE);
+
+	rmb_menu->reset_size();
+	rmb_menu->set_position(get_screen_position() + p_pos);
+	rmb_menu->popup();
+}
+
+void ObjectDBProfilerPanel::_rmb_menu_pressed(int p_tool, bool p_confirm_override) {
+	String file_path = snapshot_list->get_selected()->get_metadata(0);
+	String global_path = ProjectSettings::get_singleton()->globalize_path(file_path);
+	switch (rmb_menu->get_item_id(p_tool)) {
+		case RC_MENU_OPERATIONS::SHOW_IN_FOLDER: {
+			OS::get_singleton()->shell_show_in_file_manager(global_path, true);
+			break;
+		}
+		case RC_MENU_OPERATIONS::DELETE: {
+			DirAccess::remove_file_or_error(global_path);
+			snapshot_list->get_root()->remove_child(snapshot_list->get_selected());
+			snapshot_list->set_selected(snapshot_list->get_root()->get_first_child());
+			_update_diff_items();
+			break;
+		}
+		case RC_MENU_OPERATIONS::RENAME: {
+			snapshot_list->edit_selected(true);
+			break;
+		}
+	}
+}
+
+void ObjectDBProfilerPanel::_edit_snapshot_name() {
+	const String &new_snapshot_name = snapshot_list->get_selected()->get_text(0);
+	const String &full_file_with_path = snapshot_list->get_selected()->get_metadata(0);
+	Vector<String> full_path_parts = full_file_with_path.rsplit("/", false, 1);
+	const String &full_file_path = full_path_parts.get(0);
+	const String &file_name = full_path_parts.get(1);
+	const String &old_snapshot_name = file_name.split(".").get(0);
+	const String &new_full_file_path = full_file_path.path_join(new_snapshot_name) + ".odb_snapshot";
+
+	bool name_taken = false;
+	for (int i = 0; i < snapshot_list->get_root()->get_child_count(); i++) {
+		TreeItem *item = snapshot_list->get_root()->get_child(i);
+		if (item != snapshot_list->get_selected()) {
+			if (item->get_text(0) == new_snapshot_name) {
+				name_taken = true;
+				break;
+			}
+		}
+	}
+
+	if (name_taken || new_snapshot_name.contains(":") || new_snapshot_name.contains("\\") || new_snapshot_name.contains("/") || new_snapshot_name.begins_with(".") || new_snapshot_name.length() == 0) {
+		EditorNode::get_singleton()->show_warning(TTR("Invalid snapshot name"));
```

Comment:
Use sentence case for dialogs:

```suggestion
		EditorNode::get_singleton()->show_warning(TTR("Invalid snapshot name."));
```